### DISCUSSION TITLE
Cast of _id in aggregate pipeline

### DISCUSSION THREAD

**Discussion by kunalindoliya** [Automattic/mongoose#13554] [General]
Mongoose doesn't cast the _id string to ObjectId in $match of aggregation pipeline. Wouldn't it be better to add this functionality ?

**Comment by vkarpov15** [comment]
The problem is that the aggregation framework can transform the shape of documents, so `_id` in `$match` might not be the same type as what the schema defines. We may be able to cast `_id` in the limited case where `$match` is the first stage in the aggregation pipeline, but we figure it is a cleaner API to just say Mongoose doesn't cast aggregation pipelines, rather than explain in what cases Mongoose casts aggregation pipelines.
